_posts/2020-10-19-angular-tutorial-deutsch/2020-10-19-angular-tutorial-deutsch.md

```diff
@@ -0,0 +1,447 @@
+
+> **Klassen** wurden in ES2015 eingeführt, um Konzepte wie unter anderem Vererbung und Konstruktoren nicht mehr über Prototypen abbilden zu müssen. Diese können nun über eine einfache und saubere Syntax erstellt werden.
+
+> **Decorator** sind strukturierte Metadaten einer Klasse. Ihr kennt diese vielleicht aus anderen Programmiersprachen wie z.B. Java. Das eigentliche fachliche Verhalten der Komponente bilden wir innerhalb der Klasse mit Methoden ab.
+
+Eine Komponenten-Definition besteht primär aus folgenden Bestandteilen:
+
+
+* Einem **Component-Decorator** welcher die Komponente innerhalb von Angular bekannt macht.
+* Einer **Selektor** welcher das HTML-Element beschreibt welches wir erzeugen
+* Einem **HTML-Template** welches die Darstellung unserer Komponente definiert
+* Einer **Klasse** welche das Interface und die Anzeige-Logik der Komponente beschreibt.
+
+<img class="img-fluid img-rounded" src="first-component-class-example.png" alt="Example of our first ToDo Item Component">
+
+Unsere erste Komponente wird eine statische Infobox sein. Um diese zu generieren nutzen wir wieder die Angular-CLI.
+Ihr könnt hierzu ein neuen Terminal öffnen oder den laufenden `ng serve` kurzzeitig stoppen.
+Der Serve-Prozess erkannt aber automatisch Veränderungen innerhalb eures Quellcode und kompiliert die jeweils aktuelle Version ihrer Anwendung in wenigen Sekunden.
+Ich würde euch also empfehlen einen zweiten Terminal zu öffnen und folgenden Befehl zu benutzen.
+
+```bash
+$ ng generate component info-box
+CREATE src/app/info-box/info-box.component.scss (0 bytes)
+CREATE src/app/info-box/info-box.component.html (23 bytes)
+CREATE src/app/info-box/info-box.component.spec.ts (636 bytes)
+CREATE src/app/info-box/info-box.component.ts (277 bytes)
+UPDATE src/app/app.module.ts (0 bytes)
+```
+
+Die für uns aktuell relevanten Dateien sind zur Zeit die `info-box.component.ts` und unser Template `info-box.component.html`. Schauen wir uns zunächst einmal unsere Klasse an.
+
+``` typescript
+@Component({
+ selector: 'app-info-box',
+ templateUrl: './info-box.component.html',
+ styleUrls: ['./info-box.component.scss']
+})
+export class InfoBoxComponent implements OnInit {
+
+ constructor() { }
+
+ ngOnInit() {
+ }
+
+}
+```
+
+Hier sehen wir wie erwartet eine Komponente. Unser Selektor hat den automatischen Prefix `app-` bekommen. Somit ist unsere neue Komponente nun unter dem HTML-Tag `<app-info-box></app-info-box>` nutzbar. Der Einstiegspunkt unser kompletten Anwendung ist ebenfalls eine Komponente mit dem Namen `AppComponent`.
+Um unsere frisch generierte Komponente anzuzeigen, müssen wir diese in dem Template unser Anwendung aufrufen. Hierzu geht ihr in die Datei `app.component.html`, löscht dort den kompletten derzeitigen Inhalt und fügt eure Komponente via HTML-Tag ein.
+
+```html
+<app-info-box></app-info-box>
```

Comment:
```suggestion
<!-- app.component.html> 
<app-info-box></app-info-box>
```

Für Leute, die den Text nach Snippets scannen, ist es hilfreich in den Code-Snippets die Dateinamen im Kommentar zu nennen.
Damit können die Snippets schnell zugeordnet werden.